shell/components/form/NodeAffinity.vue

```diff
@@ -50,7 +51,10 @@ export default {
       return {
         allSelectorTerms,
         weightedNodeSelectorTerms: preferredDuringSchedulingIgnoredDuringExecution,
-        defaultWeight:             1
+        defaultWeight:             1,
+        // rules in MatchExpressions.vue can not catch changes what happens on parent component
+        // we need re-render it via key changing
+        rerenderNums:              0
```

Comment:
Just tested your code and this is not needed... The secret to your fix lies on the `@remove` event at the `ArrayListGrouped` component. All references to `rerenderNums` should be deleted from the PR